conduit/pipeline/pipeline.go

```diff
@@ -394,6 +408,13 @@ func (p *pipelineImpl) Start() {
 		for {
 		pipelineRun:
 			metrics.PipelineRetryCount.Observe(float64(retry))
+			if uint64(retry) >= p.cfg.RetryCount {
+				p.logger.Errorf("Pipeline has reached maximum retry count (%d) - stopping...", p.cfg.RetryCount)
+				return
+			}
+			if retry > 0 {
+				time.Sleep(p.cfg.RetryDelay)
+			}
```

Comment:
What is the expected behavior if RetryCount is 0?

Reply:
If it is zero then there is no delay.  Might be useful for some stress testing applications

Reply:
I meant the `RetryCount` field, looks like we return without trying even once

Reply:
Fixed and tests added